### PR TITLE
Make FOG_ENABLED a shader_def instead of material flag

### PR DESCRIPTION
**Pull request by IceSentry**
# Objective

- If the fog is disabled it still generates a useless branch which can hurt performance

## Solution

- Make the flag a shader_def instead

## Testing

- I tested enabling/disabling fog works as expected per-material in the fog example
- I also tested that scenes that don't add the FogSettings resource still work correctly

## Review notes

I'm not sure how to handle the removed material flag. Right now I just commented it out and added a not to reuse it instead of creating a new one.

### CODE REVIEW

**Review by robtfm — APPROVED** [review]
makes sense, fog tends to be all or nothing so this is unlikely to make more shader variants in any real world scenarios, and saves some registers when not used.

i think keeping the other flags unchanged and commenting the old fog flag as unused is the right call as well.

File: crates/bevy_pbr/src/pbr_material.rs [hunk 1]
```diff
@@ -867,7 +867,7 @@ bitflags::bitflags! {
         const UNLIT                      = 1 << 5;
         const TWO_COMPONENT_NORMAL_MAP   = 1 << 6;
         const FLIP_NORMAL_MAP_Y          = 1 << 7;
-        const FOG_ENABLED                = 1 << 8;
+        // const UNUSED                  = 1 << 8; // USE THIS IF YOU ADD A NEW FLAG
```
Comment (coreh) [inline crates/bevy_pbr/src/pbr_material.rs:870]: Is the concern here that renumbering everything might conflict with other open PRs?

**Review by coreh — APPROVED** [review]

File: crates/bevy_pbr/src/pbr_material.rs (hunk 1 above)
Comment (IceSentry) [reply crates/bevy_pbr/src/pbr_material.rs:870]: I mostly just wasn't sure if that was the way to do it or not 😅

**Review by JMS55 — APPROVED** [review]

**Comment by pcwalton** [comment]
This wasn't done the right way, I don't think. What I suggested was to make the `FOG_ENABLED` shader def contingent on whether the view has fog, not whether the material had a flag that disables it.

**Pull request merged**